### ISSUE TITLE
`ng new` does not show help from custom schematic

### ISSUE THREAD

**Issue by vakrilov** [angular/angular-cli#13681]
# 🐞 Bug report

### Command (mark with an `x`)

```
- [x] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?

 Yes, the previous version in which this bug was not present was: `6.1.3`

### Description
When executing:
```
ng new -c=<custom-collection> --help
```
The help displayed is **not** the help from the `ng-new` schematic in the custom collection. The help from `@schematics/angular` is displayed instead.

## 🔬 Minimal Reproduction
I have reproduced it with `@nativescript/schematis` package:

```
npm i -g @nativescript/schematics
ng new -c=@nativescript/schematics --help
```
**Expected:** Print the help for `@nativescript/schematics` ng-new
**Actual:** Print the help for `@schematics/angular` 
**Note:** Using `@angular/cli@6.1.3` prints the correct help

## 🌍 Your Environment
<pre><code>
Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.13.1
@angular-devkit/core         7.3.1
@angular-devkit/schematics   7.3.1
@schematics/angular          7.3.1
@schematics/update           0.13.1
rxjs                         6.3.3
typescript                   3.2.4
</code></pre>

**Anything else relevant?**
Debugging the issue led me to [this place in the angular cli code](https://github.com/angular/angular-cli/blob/master/packages/angular/cli/models/schematic-command.ts#L103-L113). `schematicName` is defaulted to `ng-new` when executing `ng new`. However, the `options.collection`(if such option is provided) should used to load the collection schema and description.

Issue was reported here also: https://github.com/NativeScript/nativescript-schematics/issues/157

**Comment by cexbrayat** [comment]
@alan-agius4 You probably fixed it with #14530 right?

**Comment by alan-agius4** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent Angular CLI version.

If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.
